backend/src/bundles/users/user.service.ts

```diff
@@ -49,10 +53,48 @@ class UserService implements Service {
     }
 
     public async update(
+        query: Record<string, number>,
```

Comment:
probably would be better to change it to updateProfileById or so I don't think that we will update by another criteria

Reply:
It was the simple `id` instead of `query`, but as `updatePassword` also use Repository type update I changed it to the `query` for mine implementation too, also I think I can't rename this method to `updateProfileById` because the service.type has only update method. And if I add a new one, it will ask to implement this method all through the other services in the app

```
type Service<T = unknown> = {
    find(query: Record<string, T>): Promise<T>;
    
    findAll(): Promise<{
        items: T[];
    }>;
    
    create(payload: unknown): Promise<T>;
    
    update(
        query: Record<string, T>,
        payload: Record<string, unknown>,
    ): Promise<T | null>;
    
    delete(payload: unknown): Promise<boolean>;
};
```

Reply:
You don't need to add it to the service type, these annoyations here mostly to enforce naming conventions for common methods.